conans/model/options.py

```diff
@@ -160,21 +163,38 @@ def __init__(self, values=None):
         if not values:
             return
 
+        assert not isinstance(values, six.string_types), "String type not expected in OptionValues " \
+                                                         "constructor, did you mean to call " \
+                                                         "OptionValues.loads(<string>)"
+
         # convert tuple "Pkg:option=value", "..." to list of tuples(name, value)
         if isinstance(values, tuple):
-            new_values = []
-            for v in values:
-                option, value = v.split("=", 1)
-                new_values.append((option.strip(), value.strip()))
-            values = new_values
+            values = [item.split("=", 1) for item in values]
+
+        # convert dict {"Pkg:option": "value", "..": "..", ...} to list of tuples (name, value)
+        if isinstance(values, dict):
+            values = [(k, v) for k, v in values.items()]
+
+        # Check that we have a list of tuples
+        offenders = list(filter(lambda u: not isinstance(u, (tuple, list)) or len(u) <= 1, values))
```

Comment:
overly protective/defensive, not needed. I'd try to apply the "python zen" whenever possible. Also, do not check types and do defensive programming unless there is strong evidence from users/usage that it is needed because of frequent error.

In python, ``a = {1: 2, 1: 2}`` is valid, won't raise an error.

This helps to keep the code as simple as possible, future maintenance is very important.

Reply:
Here (L179) I implement in just a single point that every option has at least a value and build a meaningful message to the user. I don't see where is the over protection. But we should better talk about it here (https://github.com/conan-io/conan/pull/3424#pullrequestreview-151964499).

About the dict you propose (with non-string keys) it will raise a weird error in line L192, but I'm not protecting against it because I think that an option name will always be a string.

Reply:
I still don't see it. If necessary, convert dict entries to strings on input, instead of checking them and raising, it would be more consistent to the current approach.

Reply:
These lines of code come from the PR #3424 which has to be merged first. And the lines you don't like are already removed. I will merge that branch now to update it here.